From 08163c2e044741179ad40fc8bfa214c5149989ef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alejandro=20Pi=C3=B1eiro?= <apinheiro@igalia.com>
Date: Thu, 29 May 2014 14:12:16 +0200
Subject: [PATCH] a11y: initialize atspi on demand

Only call atspi.init if needed. This is also more
coherent with the listener registration, that is
only done when needed.

https://bugzilla.gnome.org/show_bug.cgi?id=730118

Conflicts:
	js/ui/focusCaretTracker.js
---
 js/ui/focusCaretTracker.js | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/js/ui/focusCaretTracker.js b/js/ui/focusCaretTracker.js
index 5af8d37..576a0ca 100644
--- a/js/ui/focusCaretTracker.js
+++ b/js/ui/focusCaretTracker.js
@@ -32,9 +32,9 @@ const FocusCaretTracker = new Lang.Class({
     Name: 'FocusCaretTracker',
 
     _init: function() {
-        Atspi.init();
-        Atspi.set_timeout(250, 250);
         this._atspiListener = Atspi.EventListener.new(Lang.bind(this, this._onChanged));
+
+        this._atspiInited = false;
     },
 
     _onChanged: function(event) {
@@ -44,12 +44,24 @@ const FocusCaretTracker = new Lang.Class({
             this.emit('caret-moved', event);
     },
 
+    _initAtspi: function() {
+        if (!this._atspiInited) {
+            Atspi.init();
+            Atspi.set_timeout(250, 250);
+            this._atspiInited = true;
+        }
+    },
+
     registerFocusListener: function() {
+        this._initAtspi();
+
         return this._atspiListener.register(STATECHANGED + ':focused') &&
                this._atspiListener.register(STATECHANGED + ':selected');
     },
 
     registerCaretListener: function() {
+        this._initAtspi();
+
         return this._atspiListener.register(CARETMOVED);
     },
 
-- 
2.0.0

